parlai/core/metrics.py

```diff
@@ -352,7 +376,7 @@ def update(self, observation, labels):
                             # no need to lock because not SharedTable
                             if k not in self.metrics:
                                 self.metrics[k] = v
-                                self.metrics_list.append(k)
+                                self.metrics_list.add(k)
```

Comment:
i don't understand this one, I don't see why custom metrics should be polluting the Teacher's metric computation list.

Reply:
I think I just keep it as it is? or you suggest sth else?